src/backend/dx12/src/lib.rs

```diff
@@ -688,55 +515,255 @@ impl hal::Instance for Instance {
                 unsafe { ComPtr::new(adapter as *mut dxgi1_2::IDXGIAdapter2) }
             };
 
+            cur_index += 1;
+
             // Check for D3D12 support
-            let hr = unsafe {
-                d3d12::D3D12CreateDevice(
-                    adapter.as_mut() as *mut _ as *mut _,
-                    d3dcommon::D3D_FEATURE_LEVEL_11_0, // TODO: correct feature level?
-                    &d3d12::IID_ID3D12Device,
-                    ptr::null_mut(),
-                )
+            // Create temporaty device to get physical device information
+            let device = {
+                let mut device = ptr::null_mut();
+                let hr = unsafe {
+                    d3d12::D3D12CreateDevice(
+                        adapter.as_mut() as *mut _ as *mut _,
+                        d3dcommon::D3D_FEATURE_LEVEL_11_0,
+                        &d3d12::IID_ID3D12Device,
+                        &mut device as *mut *mut _ as *mut *mut _,
+                    )
+                };
+                if !winerror::SUCCEEDED(hr) {
+                    continue;
+                }
+
+                unsafe { ComPtr::<d3d12::ID3D12Device>::new(device) }
             };
 
-            if winerror::SUCCEEDED(hr) {
-                // We have found a possible adapter
-                // acquire the device information
-                let mut desc: dxgi1_2::DXGI_ADAPTER_DESC2 = unsafe { std::mem::uninitialized() };
-                unsafe { adapter.GetDesc2(&mut desc); }
-
-                let device_name = {
-                    let len = desc.Description.iter().take_while(|&&c| c != 0).count();
-                    let name = <OsString as OsStringExt>::from_wide(&desc.Description[..len]);
-                    name.to_string_lossy().into_owned()
-                };
+            // We have found a possible adapter
+            // acquire the device information
+            let mut desc: dxgi1_2::DXGI_ADAPTER_DESC2 = unsafe { std::mem::uninitialized() };
```

Comment:
nit: let's save ourselves a possible headache and just use `mem:Lzeroed`
also note that `std::mem` is already imported